source/s3_util.c

```diff
@@ -488,6 +488,86 @@ int aws_s3_parse_content_length_response_header(
     return result;
 }
 
+int aws_s3_parse_request_range_header(
+    struct aws_http_headers *request_headers,
+    bool *out_initial_message_has_start_range,
+    bool *out_initial_message_has_end_range,
+    uint64_t *out_initial_start_range,
+    uint64_t *out_initial_end_range) {
+
+    AWS_PRECONDITION(request_headers);
+    AWS_PRECONDITION(out_initial_message_has_start_range);
+    AWS_PRECONDITION(out_initial_message_has_end_range);
+    AWS_PRECONDITION(out_initial_start_range);
+    AWS_PRECONDITION(out_initial_end_range);
+
+    bool initial_message_has_start_range = false, initial_message_has_end_range = false;
+    uint64_t initial_start_range = 0, initial_end_range = 0;
+
+    struct aws_byte_cursor range_header_value;
+
+    if (aws_http_headers_get(request_headers, g_range_header_name, &range_header_value)) {
+        return aws_raise_error(AWS_ERROR_S3_INVALID_RANGE_HEADER);
+    }
+
+    struct aws_byte_cursor range_header_start = aws_byte_cursor_from_c_str("bytes=");
+
+    int result = AWS_OP_ERR;
+    /* verify bytes= */
+    if (aws_byte_cursor_starts_with(&range_header_value, &range_header_start)) {
```

Comment:
utterly trivial: early out, instead of indenting the rest of the function

```suggestion
    /* verify bytes= */
    if (!aws_byte_cursor_starts_with(&range_header_value, &range_header_start)) {
        goto done;
    }
```

Reply:
Thanks, I have refactored this function to simply return, as it doesn't perform any cleanup.